### PR TITLE
Narrow down typos ignores

### PR DESCRIPTION
**Pull request by szepeviktor**
- remove ignored files
- check dotfiles
- reorder config sections: files, regexp, words, IDs
- remove non-existent ignores

### CODE REVIEW

**Comment by szepeviktor** [comment]
💡 It would be nice to rename plural Latin variables names.

- trivias
- datas
- and maybe selfs

File: .typos.toml [hunk 1]
```diff
@@ -1,31 +1,30 @@
...
-    ".*\\$\\d.*",
-    # ignore generated content like `boxed....nner()`, `Defaul...efault`
-    "\\w*\\.{3,4}\\w*",
-]
-
-[files]
-extend-exclude = [
-  "*.json",
-  "*.rast",
-  "crates/parser/test_data/lexer/err/*",
```
Comment (Young-Flash) [inline .typos.toml:29]: these are auto generated, IMO they should be exclude, since other typos occur here in the furture

Comment (szepeviktor) [reply .typos.toml:29]: Reverted.

File: .typos.toml [hunk 2]
```diff
@@ -1,31 +1,30 @@
...
+
+[default]
+extend-ignore-re = [
+    # ignore string which contains $0, which is used widely in tests
+    ".*\\$0.*",
+    # ignore generated content like `boxed....nner()`, `Defaul...efault`
+    "\\w*\\.{3,4}\\w*",
+    '"flate2"',
+    'IDENT "SER"',
+    "raison d'être",
```
[the remark this answers is not in the document]
Comment (szepeviktor) [reply .typos.toml:17]: words and identifiers couldn't help here, e.g. adding `SER` to identifiers will hide future typos of "SET"

File: .typos.toml [hunk 3]
```diff
@@ -1,31 +1,32 @@
...
-
-[files]
-extend-exclude = [
-  "*.json",
-  "*.rast",
-  "crates/parser/test_data/lexer/err/*",
-  "bench_data/*",
-]
+[default.extend-identifiers]
+0onoDigit = "0onoDigit"
```
Comment (lnicola) [inline .typos.toml:29]: I think this is excluded now?

Comment (szepeviktor) [reply .typos.toml]: Yes.
Removed.

File: .typos.toml [hunk 4]
```diff
@@ -1,31 +1,31 @@
...
+ignore-hidden = false
+
+[default]
+extend-ignore-re = [
+    # ignore string which contains $0, which is used widely in tests
+    ".*\\$0.*",
+    # ignore generated content like `boxed....nner()`, `Defaul...efault`
+    "\\w*\\.{3,4}\\w*",
+    '"flate2"',
+    'IDENT "SER"',
```
Comment (lnicola) [inline .typos.toml:16]: Same for this one? And we could also ignore `crates/project-model/test_data/*`, I don't think there's anything useful in there.

And would you mind squashing at the end?

Comment (szepeviktor) [reply .typos.toml]: As you wish.
Now comes the squashing ...

**Comment by lnicola** [comment]
Thanks!

@bors r+

**Comment by szepeviktor** [comment]
Glad to contribute.

**Pull request merged**